Gems/ROS2/Code/Source/Sensor/SensorConfiguration.h

```diff
@@ -0,0 +1,36 @@
+/*
+* Copyright (c) Contributors to the Open 3D Engine Project.
+* For complete copyright and license terms please see the LICENSE at the root of this distribution.
+*
+* SPDX-License-Identifier: Apache-2.0 OR MIT
+*
+*/
+
+#pragma once
+
+#include <AzCore/RTTI/RTTI.h>
+#include <AzCore/Serialization/SerializeContext.h>
+
+namespace ROS2
+{
+    /// General configuration for sensors
+    struct SensorConfiguration
+    {
+    public:
+        AZ_RTTI(SensorConfiguration, "{4755363D-0B5A-42D7-BBEF-152D87BA10D7}");
+
+        SensorConfiguration() = default;
+        virtual ~SensorConfiguration() = default;
+
+        static void Reflect(AZ::ReflectContext* context);
+
+        // TODO - publishing-related data
+        AZStd::string m_topic; // TODO - apply namespace, default to standard names per type, validation
```

Comment:
One thing I just thought about.

Filling up the `topic` variable and `namespace` (somewhere else) might confuse the user. I imagine that the user might just put the whole expected topic in the `topic` var (i.e. `/sensors/front/lidar`) and then be confused about `namespace`. Should they just put the `/sensors/front/` (which is de-facto a namespace), leave it blank, or rewrite the topic to `lidar`? The user doesn't have to know that namespace is going to be prepended to the topic.

Do you think that we lack some explicit hints for the user, or should we think about forbidding slashes in the `m_topic` to make it true "namespaceless" :smile: ? :thinking:

Reply:
This will be handled in ROS2 Frame where the user would be able to input a namespace. Note that namespaces are not handled yet, in hierarchy, options to generate from entity name and so on.

Reply:
I think it will be useful to implement read-only field in the Editor component to display the topic as it is (constructed from namespaces). This can improve UX.